portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+/* Shows about text when the "About" button is pressed. */
+function showAbout () {
+    const aboutContainer = document.getElementById('aboutSpace');
+    aboutTest = aboutContainer.getElementsByTagName('p');
+    if (aboutTest.length > 0) {
+        aboutText = aboutContainer.getElementsByTagName('p')[0];
+    }
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+    // Change the text in the aboutSpace element 
+    if (aboutText.className === "about-hide") {
+        aboutText.className = "about-show";
+        aboutContainer.className = "background";
+    }
+    
+    else {
+        //aboutContainer.innerText = "";
+        aboutText.className = "about-hide";
+        aboutContainer.className = "";
+    }
+}
+
+/* Shows pictures when the "Pictures" button is pressed. */
+function showPics() {
+    const imgs = ["ucla1.JPG", "ucla2.jpg.JPG", "book.JPG", "harrypotter.JPG"]
+    const srcBase = "https://8080-0ba309ed-2f3c-4b55-83b6-7741ff94bec3.us-west1.cloudshell.dev/base"
+    const picPath = "https://8080-0ba309ed-2f3c-4b55-83b6-7741ff94bec3.us-west1.cloudshell.dev/images/"
+    
+    // Calculate a random index for the img array 
+    const randInt = (Math.floor(Math.random() * imgs.length));
+    const randImg = imgs[randInt];
+    
+    const div = document.getElementById('picSpace');
+    const imgArr = div.getElementsByTagName('img');
+
```

Comment:
It's a good practice to fail "gracefully" when an element isn't found, so I think it would be good to declare `picContainer` outside of the `if` body and then check if it was found. Something like this:

```suggestion
let picContainer;  // Will be initialized to `undefined` by default.
// existing if block goes here
if (picContainer) { // Will be considered true if it has an actual element as its value, and false if it's still undefined.
  // existing code that works on picContainer goes here
}
```
Another option is to move all of the code that works on `picContainer` inside of the `if (imgArr.length > 0)` block.

Reply:
@megankj